src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsAssemblyRewriter.cs

```diff
@@ -504,7 +504,7 @@ MethodDefinition GetUnmanagedCallersOnlyAttributeConstructor (IAssemblyResolver
 				throw new InvalidOperationException ("[{targetArch}] Unable to find the System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute type");
 			}
 
-			foreach (MethodDefinition md in unmanagedCallersOnlyAttribute.Methods) {
+			foreach (MethodDefinition md in unmanagedCallersOnlyAttribute!.Methods) {
```

Comment:
Check if `unmanagedCallersOnlyAttribute` is null and throw an appropriate `ArgumentNullException`. Do this instead of using `!`.